filebeat/input/filestream/config.go

```diff
@@ -142,3 +144,78 @@ func (c *config) Validate() error {
 
 	return nil
 }
+
+// ValidateInputIDs checks all filestream inputs to ensure all have an ID and
+// the ID is unique. If there is any empty or duplicated ID, it logs an error
+// containing the offending input configurations and returns an error containing
+// the duplicated IDs.
+func ValidateInputIDs(inputs []*conf.C, logger *logp.Logger) error {
+	ids := make(map[string][]*conf.C)
+	var duplicates []string
+	var empty []*conf.C
+	for _, input := range inputs {
+		fsInput := struct {
+			ID   string `config:"id"`
+			Type string `config:"type"`
+		}{}
+		err := input.Unpack(&fsInput)
+		if err != nil {
+			return fmt.Errorf("failed to unpack filestream input configuration: %w", err)
+		}
+		if fsInput.Type == "filestream" {
+			if fsInput.ID == "" {
+				empty = append(empty, input)
+				continue
+			}
+			ids[fsInput.ID] = append(ids[fsInput.ID], input)
+			if len(ids[fsInput.ID]) > 1 {
```

Comment:
I don't see any value in collecting the same ID multiple times. If we have the same ID used in 6 different inputs it would end up in the error message 5 times.

```suggestion
			if len(ids[fsInput.ID]) == 2 {
```